apps/synchronize/src/github/synchronizer.js

```diff
@@ -135,6 +138,40 @@ export class GitHubSynchronizer {
     return { repositories, organizations };
   }
 
+  async synchronizePurchase(accountId) {
+    try {
+      const { data: payload } =
+        await this.octokit.apps.getSubscriptionPlanForAccount({
+          account_id: accountId,
+        });
+      const newPlan = await getNewPlanOrThrow(payload);
+      const account = await getOrCreateAccount(payload);
+      const activePurchase = await account.getActivePurchase();
+      if (activePurchase.planId === payload.marketplace_purchase.plan.id) {
+        return;
+      }
+
+      transaction(async (trx) => {
+        await Promise.all(() => [
+          Purchase.query(trx)
+            .patch({ endDate: payload.effective_date })
+            .findById(activePurchase.id),
+          Purchase.query(trx).insert({
+            accountId: account.id,
+            planId: newPlan.id,
+            startDate: payload.effective_date,
+          }),
+        ]);
```

Comment:
The code is duplicated. The logic is the same as "purchased" and "cancelled", so the code should be refactored and used both in events and in sync.